src/live.rs

```diff
@@ -59,30 +63,52 @@ pub fn iso_ignition_embed(config: &IsoIgnitionEmbedConfig) -> Result<()> {
         }
     };
 
-    let cpio = make_cpio(&ignition)?;
-    if cpio.len() > embed.length {
-        bail!(
-            "Compressed Ignition config is too large: {} > {}",
-            cpio.len(),
-            embed.length
-        );
-    }
-    if !config.force {
-        // Ensure all zero bytes
-        embed.seek_to_start()?;
-        let mut buf = embed.new_buffer();
-        embed.read(&mut buf)?;
-        // compare to zeroed buffer
-        if buf != embed.new_buffer() {
-            bail!("This ISO image already has an embedded Ignition config; use -f to force.");
+    if config.output.as_deref() == USE_STDOUT {
+        let mut file = OpenOptions::new()
+            .read(true)
+            .open(&config.input)
+            .chain_err(|| format!("opening {}", &config.input))?;
+        let mut embed = EmbedArea::for_file(&mut file)?;
+
+        let cpio = make_cpio(&ignition)?;
+        if cpio.len() > embed.length {
+            bail!(
+                "Compressed Ignition config is too large: {} > {}",
+                cpio.len(),
+                embed.length
+            );
+        }
+
+        embed.write_to_stdout(&cpio)?;
+    } else {
+        let mut holder = CopiedFileHolder::new(&config.input, config.output.as_ref())?;
+        let mut embed = EmbedArea::for_file(&mut holder.file)?;
+
+        let cpio = make_cpio(&ignition)?;
+        if cpio.len() > embed.length {
+            bail!(
+                "Compressed Ignition config is too large: {} > {}",
+                cpio.len(),
+                embed.length
+            );
         }
+        if !config.force {
```

Comment:
Let's do this check in the stdout path also.  It's a bit odd, but it makes the semantics more consistent.  Note that we currently do the check with `-o`, which is in a similar situation.